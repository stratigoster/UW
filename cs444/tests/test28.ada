PACKAGE main IS
  TYPE data;
  TYPE pdata is access data;
  d : data;
--    ^^^^
-- type `data' is not fully defined yet
  TYPE data IS RECORD
    a : integer;
    b : boolean;
    p : pdata;
    g : grades;
--      ^^^^^^
-- reference to undefined type `grades'
  END RECORD;
END main;

PACKAGE BODY main IS
BEGIN
  null;
END main;
