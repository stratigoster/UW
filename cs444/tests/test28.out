***ERROR***:4:12:	Missing full type declaration for ` data ' before use
***ERROR***:11:16:	Uninitialized type ` grades '

------
Compilation error
------
